### PR TITLE
objects in bounding box 2d implementation + new tests for 2D_modeler

### PR DESCRIPTION
**Pull request by gmalinve**
In this PR:
- a new method to check whether a 2d object is in a bounding box or not has been implemented.
- new test in modeler 2d have been added + old methods in Model2D

### CODE REVIEW

File: pyaedt/modeler/Model2D.py [hunk 1]
```diff
@@ -177,3 +177,57 @@ def radial_split_2D(self, radius, name):
...
+    @pyaedt_function_handler()
+    def objects_in_bounding_box(self, bounding_box, check_lines=True, check_sheets=True):
+        """Given a 2D bounding box checks if sheets and lines are inside it.
+
+        Parameters
+        ----------
+        bounding_box : list.
+            List of either 4 or 6 coordinates of bounding box.
+        check_lines : bool, optional.
+            Check line objects.
```
Comment (PipKat) [inline pyaedt/modeler/Model2D.py:190]:
```suggestion
            Whether to check line objects. The default is ``True``.
```

File: pyaedt/modeler/Model2D.py [hunk 2]
```diff
@@ -177,3 +177,57 @@ def radial_split_2D(self, radius, name):
...
+        check_sheets : bool, optional.
+            Check sheet objects.
```
Comment (PipKat) [inline pyaedt/modeler/Model2D.py:192]:
```suggestion
            Whether to check sheet objects. The default is ``True``.
```

File: pyaedt/modeler/Model2D.py [hunk 3]
```diff
@@ -177,3 +177,57 @@ def radial_split_2D(self, radius, name):
...
+
+        Returns
+        -------
+        list of :class:`pyaedt.modeler.Object3d`
+        """
+
+        if len(bounding_box) != 4:
+            raise ValueError("Bounding box list must have dimension 4.")
```
Comment (PipKat) [inline pyaedt/modeler/Model2D.py:200]: Should this be written as is or rather as "Bounding box list must have four dimensions."

File: pyaedt/modeler/Model2D.py [hunk 4]
```diff
@@ -177,3 +177,57 @@ def radial_split_2D(self, radius, name):
...
         return True
+
+    @pyaedt_function_handler()
+    def objects_in_bounding_box(self, bounding_box, check_lines=True, check_sheets=True):
+        """Given a 2D bounding box checks if sheets and lines are inside it.
+
+        Parameters
+        ----------
+        bounding_box : list.
+            List of either 4 or 6 coordinates of bounding box.
```
Comment (PipKat) [inline pyaedt/modeler/Model2D.py:188]:
```suggestion
            List of either the 4 or 6 coordinates of the bounding box.
```

File: pyaedt/modeler/Model2D.py (hunk 3 above)
Comment (gmalinve) [reply pyaedt/modeler/Model2D.py:200]: I meant to say that bounding box list must be a list of 4 elements.
Any suggestion is welcome!

**Review by maxcapodi78 — APPROVED** [review]

**Pull request merged**